cpp/CMakeLists.txt

```diff
@@ -138,6 +138,7 @@ set(CMAKE_EXE_LINKER_FLAGS "-Wl,--disable-new-dtags")
 option(BUILD_SHARED_LIBS "Build shared libraries" ON)
 option(BUILD_TESTS "Configure CMake to build tests" ON)
 option(BUILD_BENCHMARKS "Configure CMake to build (google) benchmarks" OFF)
+option(BUILD_CUDATASOURCE "Configure CMake to build cuDatasource" OFF)
```

Comment:
Similarly, this shouldn't be called `cuDataSource`. We'd want a separate library for each integration, so lets name it as such with `CUDF_KAFKA` or something similar.